.github/workflows/build.yaml

```diff
@@ -3,116 +3,372 @@ on: [ push, pull_request, workflow_dispatch ]
+
+      - name: Build Deployment Package
+        run: |
+          mkdir -p /tmp/node-build
+          ./build-scripts/ubuntu-1604/build-indy-node.sh "/__w/indy-node/indy-node" "${{ steps.version.outputs.upstreamVer }}" "/tmp/node-build" "${{ steps.version.outputs.pkgVer }}"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: node-deb
+          path: /tmp/node-build
+          retention-days: 5
+
+  build_3rd_party_dependencies:
+    name: Build 3rd Party Dependencies
+    # needs: [workflow-setup, build-image]
+    needs: [workflow-setup, indy_node_tests, lint]
+    runs-on: ubuntu-20.04
+    container:
+      image: ghcr.io/${{ needs.workflow-setup.outputs.GITHUB_REPOSITORY_NAME }}/node-build:ubuntu-16-04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Try load from cache.
+        id: third-party-dependencies
+        uses: actions/cache@v2
+        with:
+          path: /tmp/third-party-dependencies
+          key: third-party-dependencies-${{ hashFiles('./build-scripts/ubuntu-1604/build-3rd-parties.sh') }}
+
+      - name: Build 3rd party deployment packages
+        if: steps.third-party-dependencies.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p ./build-scripts/ubuntu-1604/cache/3rd-party-dependencies/
+          ./build-scripts/ubuntu-1604/build-3rd-parties.sh ./cache/3rd-party-dependencies
+          mv ./build-scripts/ubuntu-1604/cache/* /tmp/third-party-dependencies
+
+  publish_artifacts:
+    name: Publish Artifacts
+    runs-on: ubuntu-20.04
+    needs: [workflow-setup, build_release, build_3rd_party_dependencies]
+    if: needs.workflow-setup.outputs.publish == 'true'
+    env:
+      GITHUB_REF: ${{ needs.workflow-setup.outputs.GITHUB_REF }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Setup JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v1
+        env:
+          JF_ARTIFACTORY_1: ${{ secrets.INDY_ARTIFACTORY_REPO_CONFIG }}
+
+      - name: Ping Artifactory
+        run: |
+          # Test the connection to Ping the Hyperledger Artifactory server
+          # to ensure everything has been setup correctly.
+          jfrog rt ping
+
+      - name: Download Node Artifacts from Pipeline Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: node-deb
+          path: to_publish
+
+      - name: Publish Node Artifacts
+        uses: ./.github/actions/publish-deb
+        with:
+          sourceDirectory: /home/runner/work/indy-node/indy-node/to_publish
+          distribution: xenial
+          component: ${{ env.GITHUB_REF }}
+
+      - name: Download 3rd Party Artifacts Dependencies from Cache
+        id: third-party-dependencies
+        uses: actions/cache@v2
+        with:
+          path: /tmp/third-party-dependencies
+          key: third-party-dependencies-${{ hashFiles('./build-scripts/ubuntu-1604/build-3rd-parties.sh') }}
+          restore-keys: |
```

Comment:
restore keys should be removed because we want to explicitly publish the 3rd party dependencies of the current run. 
The job should fail if the required 3rd party dependencies are not available in the cache.

Reply:
Update